pypeit/spectrographs/keck_hires.py

```diff
@@ -55,19 +161,81 @@ def init_meta(self):
         """
         self.meta = {}
         # Required (core)
-        self.meta['ra'] = dict(ext=0, card='RA')
-        self.meta['dec'] = dict(ext=0, card='DEC')
+        self.meta['ra'] = dict(ext=0, card='RA', required_ftypes=['science', 'standard'])
+        self.meta['dec'] = dict(ext=0, card='DEC', required_ftypes=['science', 'standard'])
         self.meta['target'] = dict(ext=0, card='OBJECT')
         self.meta['decker'] = dict(ext=0, card='DECKNAME')
-        self.meta['binning'] = dict(ext=0, card='BINNING')
-
+        self.meta['binning'] = dict(card=None, compound=True)
         self.meta['mjd'] = dict(ext=0, card='MJD')
-        self.meta['exptime'] = dict(ext=0, card='EXPTIME')
+        # This may depend on the old/new detector
+        self.meta['exptime'] = dict(ext=0, card='ELAPTIME')
         self.meta['airmass'] = dict(ext=0, card='AIRMASS')
-        self.meta['dispname'] = dict(ext=0, card='ECHNAME')
+        #self.meta['dispname'] = dict(ext=0, card='ECHNAME')
         # Extras for config and frametyping
-#        self.meta['echangl'] = dict(ext=0, card='ECHANGL')
-#        self.meta['xdangl'] = dict(ext=0, card='XDANGL')
+        self.meta['hatch'] = dict(ext=0, card='HATOPEN')
+        self.meta['dispname'] = dict(ext=0, card='XDISPERS')
+        self.meta['filter1'] = dict(ext=0, card='FIL1NAME')
+        self.meta['echangle'] = dict(ext=0, card='ECHANGL', rtol=1e-3)
+        self.meta['xdangle'] = dict(ext=0, card='XDANGL', rtol=1e-3)
+#        self.meta['idname'] = dict(ext=0, card='IMAGETYP')
+        # NOTE: This is the native keyword.  IMAGETYP is from KOA.
+        self.meta['idname'] = dict(ext=0, card='OBSTYPE')
+        self.meta['frameno'] = dict(ext=0, card='FRAMENO')
+        self.meta['instrument'] = dict(ext=0, card='INSTRUME')
+
+    def compound_meta(self, headarr, meta_key):
+        """
+        Methods to generate metadata requiring interpretation of the header
+        data, instead of simply reading the value of a header card.
+
+        Args:
+            headarr (:obj:`list`):
+                List of `astropy.io.fits.Header`_ objects.
+            meta_key (:obj:`str`):
+                Metadata keyword to construct.
+
+        Returns:
+            object: Metadata value read from the header(s).
+        """
+        if meta_key == 'binning':
+            # TODO JFH Is this correct or should it be flipped?
```

Comment:
Resolve this now (i.e. in this PR)

Reply:
I think it is right but don't have time to figure it out.